scripts/lib/rules/PascalCase.js

```diff
@@ -29,9 +29,19 @@ async function validate({
   publicEventsFunctions,
   onError,
 }) {
+  // Check if extension name is a valid string
+  if (typeof name !== 'string') {
+    onError('Invalid extension name. Expected a string.');
+    return;
+  }
+
   if (legacyCamelCaseExtensions.has(name)) return;
-  for (const { name } of publicEventsFunctions) checkPascalCase(name, onError);
-  for (const { name } of eventsBasedBehaviors) checkPascalCase(name, onError);
+
+  // Combine both loops into one to reduce duplication
+  const allFunctions = [...publicEventsFunctions, ...eventsBasedBehaviors];
+  for (const { name } of allFunctions) {
+    checkPascalCase(name, onError);
+  }
```

Comment:
This offers no new benefit, does an inefficient copy, and unnecessarily complexifies the logic.